plugin/pkg/admission/alwayspullimages/admission.go

```diff
@@ -101,12 +103,46 @@ func (*AlwaysPullImages) Validate(ctx context.Context, attributes admission.Attr
 	return nil
 }
 
+// check if it's update and it doesn't change the images referenced by the pod spec
+func isUpdateWithNoNewImages(attributes admission.Attributes) bool {
+	if attributes.GetOperation() == admission.Update {
+		pod, ok := attributes.GetObject().(*api.Pod)
+		if !ok {
+			klog.Warningf("Resource was marked with kind Pod but pod was unable to be converted.")
+			return false
+		}
+
+		oldPod, ok := attributes.GetOldObject().(*api.Pod)
+		if !ok {
+			klog.Warningf("Resource was marked with kind Pod but old pod was unable to be converted.")
+			return false
+		}
+
+		oldImages := sets.NewString()
+		pods.VisitContainersWithPath(&oldPod.Spec, field.NewPath("spec"), func(c *api.Container, _ *field.Path) bool {
+			oldImages.Insert(c.Image)
+			return true
+		})
+
+		hasNewImage := false
+		pods.VisitContainersWithPath(&pod.Spec, field.NewPath("spec"), func(c *api.Container, _ *field.Path) bool {
+			if !oldImages.Has(c.Image) {
+				hasNewImage = true
+			}
+			return !hasNewImage
+		})
+		return !hasNewImage
+	}
+	return false
+}
+
 func shouldIgnore(attributes admission.Attributes) bool {
 	// Ignore all calls to subresources or resources other than pods.
 	if len(attributes.GetSubresource()) != 0 || attributes.GetResource().GroupResource() != api.Resource("pods") {
 		return true
+	} else if isUpdateWithNoNewImages(attributes) {
```

Comment:
no need to chain the else since the if block returns. simplify to:

```suggestion
	}
	
	if isUpdateWithNoNewImages(attributes) {
```

Reply:
@liggitt  Thanks for your patience and detailed comments. Appreciated.